packages/qdrant-loader/src/qdrant_loader/connectors/git/connector.py

```
@@ -432,6 +431,35 @@ async def get_documents(self) -> list[Document]:
             self.logger.error("Failed to get documents", error=str(e))
             raise
 
+    async def get_documents(self) -> list[Document]:
+        """Get documents from the repository (DEPRECATED - use stream_documents)."""
+        return await super().get_documents()
+
+    async def fetch_by_id(self, entity_id: str) -> Document | None:
+        """Fetch a single file by its repo-relative path (forward slashes)."""
+        self._ensure_initialized()
+        file_path = os.path.join(self.temp_dir, *entity_id.split("/"))
+        if not os.path.exists(file_path) or not self.file_processor.should_process_file(  # type: ignore
```

Ruff (0.15.15)

[warning] Async functions should not use os.path methods, use trio.Path or anyio.path

(ASYNC240)